packages/@adobe/spectrum-css-temp/components/buttongroup/index.css

```diff
@@ -19,8 +19,10 @@ governing permissions and limitations under the License.
   align-items: flex-start;
   /* necessary so that offsetLeft on button children is correct */
   position: relative;
+  flex-grow: 1;
```

Comment:
I think this will only help if the container of the button group is a flex
Did you see it working in other cases? you can probably check by doing this to the default story `<div>{render({})}</div>`

I was thinking that maybe `display: inline-flex;` could change to `display: flex;` but that would have other side effects on layouts that this component is included in.

Maybe we leave this alone for the moment and go at it from a different angle.

We know that the inline-flex will attempt to be as wide as its children up to the horizontal space it has available. So when we switch from column to row, the inline-flex should expand a little at the very least, but for some reason, it isn't. This is what I expect to happen:
https://jsfiddle.net/snowystinger/fwxuqzs1/9/
We may be doing something at the wrong time with the dirty logic, we might try setting the flex-direction, measuring it, and potentially setting it back all in one go. Or it may be something else entirely.

Reply:
I have tried messing with the logic. Without flex growth for some reason the ResizeObserver hook is no longer called as soon as the orientation changes from horizontal to vertical, preventing hasOverflow to going back to false and thus preventing the orientation from going back to horizontal. This however does not happen when flex-growth is set to 1 and the states related to the ButtonGroup are changed accordingly.

I was under the assumption that ButtonGroup will use the inline-flex property by default for all cases. Setting flex-growth to 1 does ensure proper functionality for all the current stories.